data_steward/analytics/cdr_ops/controlled_tier_qc/check_controlled_tier_part2.py

```diff
@@ -1225,6 +1225,105 @@ def query_template(table_era):
+# 4. Wear study start dates coincide with records in observation. Date shifted in RT.
+#
+# **If check fails:**<br> 
+# * The issue `participant with multiple records` means that those participants have multiple rows in the wear_study table, which should not be possible. Investigate the issue. Start with the CR that creates the wear_study table. <br>
+# * The issue `not in person table` means that participants exist in the wear_study table that aren't in the person table, which should not be possible. Investigate the issue. Start with the CR that creates the wear_study table.<br>
+# * The issue `no primary consent` means that participants exist in the wear_study table that do not have proper primary consent. Investigate the issue. It is possible that there is another way to determine primary consent. <br>
+# * The issue `incorrect_start_date` means that the wear_study start date does not match a record in the observation table. This is most likely due to improper date shift. Investigate the issue. Start with the CR that creates the wear_study table.    <br>
+
+# +
+query = JINJA_ENV.from_string("""
+
+WITH latest_primary_consent_records AS ( -- most current consent record per person --
+    SELECT person_id, observation_source_value, MAX(observation_date) AS latest_date,
+    FROM `{{project_id}}.{{rt_cdr_deid}}.observation` o
+    WHERE REGEXP_CONTAINS(observation_source_value, '(?i)extraconsent_agreetoconsent')
+    GROUP BY person_id, observation_source_value
+)
+
+SELECT
+  'participant with multiple records' as issue,
+  COUNT(research_id) as bad_rows
+FROM `{{project_id}}.{{rt_cdr_deid}}.wear_study` ws
+GROUP BY research_id
+HAVING COUNT(research_id)>1
+
+UNION ALL
+
+SELECT
+  'not in person table' as issue,
+  COUNT(research_id) as bad_rows
+FROM `{{project_id}}.{{rt_cdr_deid}}.wear_study` ws
+WHERE research_id not in ( -- person table --
+  SELECT person_id
+  FROM `{{project_id}}.{{rt_cdr_deid}}.person` o
+  )
+  
+UNION ALL
+
+SELECT
+  'no primary consent' as issue,
+  COUNT(research_id) as bad_rows
+FROM `{{project_id}}.{{rt_cdr_deid}}.wear_study` ws
+WHERE research_id not in (  -- aou consenting participants --
+  SELECT cte.person_id
+  FROM latest_primary_consent_records cte
+    LEFT JOIN ( -- any positive primary consent --
+      SELECT *
+      FROM `{{project_id}}.{{rt_cdr_deid}}.observation` o
+      WHERE REGEXP_CONTAINS(o.observation_source_value, '(?i)extraconsent_agreetoconsent')
+      AND o.value_as_concept_id = 45877994
+    ON cte.person_id = o.person_id
+    AND cte.latest_consent_date = o.observation_date
+  WHERE o.person_id IS NOT NULL
+  )
+
+UNION ALL
+
+SELECT
+  'incorrect_start_date' as issue,
+  COUNT(research_id) as bad_rows
+FROM `{{project_id}}.{{rt_cdr_deid}}.wear_study` ws
+LEFT JOIN ( -- all wear study consent records from observation -- 
+  SELECT * 
+  FROM `{{project_id}}.{{rt_cdr_deid}}.observation`
+  WHERE LOWER(observation_source_value) = 'resultsconsent_wear') o
```

Comment:
Not a concern, but a comment: This data may not be available if we run the WEAR suppression before this check, so we need to ensure it runs in order.

Reply:
Is it not a concern? I see we have the suppression occurring at the end of the rt and ct deid cleaning classes. Like you said the data should not be in observation at the time this runs.

Should the suppression occur in the base cleaning classes? Or if the suppression location is fine I could join to the combined dataset through the pid rid table instead?

I also see that the suppression CR was placed into the ct fitbit cleaning classes. Most likely on accident. I'll need to remove that.

Reply:
Thanks, I see you moved this check